### PR TITLE
Making pipeline stage restartable.

### PR DESCRIPTION
**Pull request by jcusa**

### CODE REVIEW

**Comment by tomaslin** [comment]
If a pipeline is already in progress or completed, the pipeline stage in the context needs to be cleaned up.

**Comment by jcusa** [comment]
thanks tomas. I am not clear on what else needs to be cleaned. logging at the stage.context of the parent pipeline, there's stageDetails in there, are you referring to that? its start/end timestamps?

**Comment by tomaslin** [comment]
The executionId and executionName fields need to be cleared or the UI will link to the wrong pipeline IIRC

**Comment by tomaslin** [comment]
pipeline might not start right away since it can be within an execution window

**Comment by jcusa** [comment]
thanks. see if it is ok now?

File: orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.groovy [hunk 1]
```diff
@@ -40,4 +42,23 @@ class PipelineStage extends LinearStage {
...
+  Stage prepareStageForRestart(Stage stage) {
+    stage = super.prepareStageForRestart(stage)
+    stage.startTime = null
+    stage.endTime = null
+
+    stage.context.remove("status")
+    stage.context.remove("executionName")
+    stage.context.remove("executionId")
+
+    stage.tasks.each { com.netflix.spinnaker.orca.pipeline.model.Task task ->
```
Comment (anotherchrisberry) [inline orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.groovy]: This should probably just be:
```groovy
stage.tasks.each { Task task ->
```
with `com.netflix.spinnaker.orca.pipeline.model.Task` as an `import`

Comment (jcusa) [inline orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.groovy]: thanks. changed.

**Comment by jcusa** [comment]
if nothing else, can this be merged in? thank you!

**Pull request merged**